### DISCUSSION TITLE
ps_account dissociates itself without any notice, making all the modules connected to it unusable

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#33332] [Q&A] [answered]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hi
today I have the third website where the module prestashop account v5.4.8 which dissociated itself from prestashop, so then all the modules connected to it stop working like the one for paypal and without notice. I find out because the shopkeepers warn me that their customers are unable to pay by card.
To solve it I have to reconnect the module to prestashop addons, I have to enter all the company information again, but at the end I got an error and it is necessary to push on the button that disconnected and reconnected, after that the connection takes place without problems and all the other modules return to function

thanks

### Steps to reproduce

it happens randomly, yesterday I updated the module from version 5.4.6 and there were no problems, same operation done on 30 other sites

### PrestaShop version(s) where the bug happened

1.7.8.9

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

ps_accounts 5.4.8

### Your company or customer's name goes here (if applicable).

maofree

**Comment by Maofree** [comment]
randomly checking other sites, now I found another one which is not associated with prestashop addons, with the same problem as the other site
I also updated this form yesterday

![1](https://github.com/PrestaShop/PrestaShop/assets/653230/bd72e615-6bf0-4aec-951f-de35d9a3e0f5)

I think there is a cookie problem from the browser, because if I manage many sites and I log into addons, this module continues to see the past login

I have to log out of the past account and then it asks me to put back all the company data on the various steps, but it's already done in the first installation

**Comment by emmanuelgautier** [comment]
Hello maofree,

After an investigation of the latest ps_accounts version, we did not notice anything wrong. The latest release brings a major update of the UI used by this module but nothing about the association. The update should not have any impact. You can find the UI Open Source project on this [repository](https://github.com/PrestaShopCorp/prestashop_accounts_vue_components).

About your last screenshot, the shop associated is associated with an account. If a shop has been associated with an account and you want to associate it again, you have to use the same account used for the first association.

Let us know if you encounter the issue again.

**Comment by o-cymoz** [comment]
Hi, 

Same problem here after update from 5.4.6 to 4.5.8

PrestaShop version(s) where the bug happened
1.7.8.8

And i'm sure that the account used is the owner.
Prestashop checkout is now disconnected and i can't recieved any payments...

Any help, or clue will be appreciated ;)

**Reply by Guillaumelpg** [reply]
Hello o-cymoz, 

We are working on a warning message to give more context in this case. Theoretically this could happen when your store tokens have expired. 

Did you have an error when you try to re-associate your store with your account?

**Reply by o-cymoz** [reply]
thx for your reply i got the same message tha, above
![prestashop_log](https://github.com/PrestaShop/PrestaShop/assets/128891037/405d8a53-a084-4cf2-9350-c86c3c1d5751)

**Reply by Guillaumelpg** [reply]
I see that your store is already associated with an account in our system.
If you log in with the owner account (webmaster@***) you should be able to unlink the store and re-associate it.

**Comment by Maofree** [comment]
Hi @o-cymoz 
try to go on prestashop addons e do the login
then go in your website and go in the ps_account module e do again the association, normally you can get an error or insert again the shop data. If you get an error, do the association more times

bye

**Reply by o-cymoz** [reply]
hi, i tried many times but association is still refused... i continue

**Comment by Maofree** [comment]
try to uninstall the module with its folders and reinstall it again

**Comment by o-cymoz** [comment]
same error... several times.
where can i find the list of my shops in the prestashop account dashboard ?
from the module, i can see two other shops but not this one. 
the problem can come from the prestashop accounts  database?

**Comment by Maofree** [comment]
go in your account on prestashop addons e check which shops there are
in same cases the shop is removed by itself

**Comment by o-cymoz** [comment]
![Capture d’écran 2023-08-03 152001](https://github.com/PrestaShop/PrestaShop/assets/128891037/cda1a355-6c14-493b-affe-1b7149dcbecf)
the shop is in the list with the two others which are associated without errors on their respective websites...

**Comment by Guillaumelpg** [comment, marked as answer]
There is a difference between the stores that are displayed on the PrestaShop Marketplace (it's only declarative) and those that are actually linked to your account via the PS Account module. 

If 3 stores are displayed on the Marketplace, this does not necessarily mean that the connected account is linked to these 3 stores.

In our system, I can clearly see the 3 stores shown in your screenshot but 2 of them are linked to an account and another is linked to another account.

If you want, you can contact me on my email address guillaume.lepage@prestashop.com so that I can give you more information about the accounts linked to your stores

**Reply by o-cymoz** [reply]
ok thx ! i'll send you an email right now.
Well if i understand, it's not a prolem with the module but perhaps my customer (i'm an agency) who made mistakes...

**Reply by Maofree** [reply]
Hi @Guillaumelpg 
about the solution that you suggest to disable the second account
If I download a website to fix some issues or to do some upgrades, normally I disable ps_account but you should consider this type of possibility, same thing if a site were placed in a module folder for testing. However, these types of causes should be removed

**Comment by o-cymoz** [comment]
It's now Ok for me, bad account associated by my customer...
Thx a lot @Guillaumelpg and @maofree for your support

**Comment by gfourquet** [comment]
Hello everyone,
We have the recurrent problem with ps_account et ps_checkout
Sometimes (every 2 days) the shops are disassociated themselves from the presta account. So we can't receive the payments.
It' so embarassing car we are are going to give up this 2 modules.
No possible to spend hours to check if it s all right

More 
When I want to reassociate the account, it's working (no password demanded because it'e recorded in the navigator I think)
When my customer wants to do the same, impossible to reassociate (And We DONT make any mistake on the pasword....).

Guillaume

**Comment by LaqueP** [comment]
This problem continues to happen in prestashop 8.0.4, I have a store associated with that version and every so often it disconnects me without prior notice, the most serious problem is that the rest of the modules such as Prestashop checkout are disconnected, with which customers are There are no payment methods left, several times we have had to delete the ps_account module by FTP and reinstall it so that it would let us associate the Prestashop account again.

We are considering even removing prestasho Checkout, since this has happened to us on weekends and we have had big drops in sales.

If Prestashop does not find a solution, we will have to hire stripe or a more reliable third-party payment gateway.

**Comment by gfourquet** [comment]
here is a small script that allows you to test the assiciation (to be callled with a crontab)

include(dirname(__FILE__).'/../../config/config.inc.php');
include(dirname(__FILE__).'/ps_accounts.php');

	$a=new Ps_accounts();
	$accountsService = $a->getService(\PrestaShop\Module\PsAccounts\Service\PsAccountsService::class);
	if ($accountsService->isAccountLinked()){
		echo 'OK';
	}else{
		echo 'KO';

**Comment by gfourquet** [comment]
I hava been in contact with prestashop support and apparently if you have a test shop connected also with this prestashop account, it disconnects the production shop.
That was our case. 
So I have desactivate the module in the test shop. 
Guillaume

**Comment by Maofree** [comment]
Hi
the question is: why is this solution not considered, given that the sites are developed, tested, corrected in development environments?
however it is not the only reason for these disconnections, because yesterday 22 sites were disconnected and my server doesn't have all those sites
if that was the reason, it should happen every day, instead it happens to me every time I update the ps_events or ps_accounts module

**Comment by Temerario74** [comment]
Same problem for me. All my shop are dissociates. How i can resolve this problem?

**Comment by Temerario74** [comment]
This is the error
![screen](https://github.com/PrestaShop/PrestaShop/assets/11271097/f5c86fd5-3e44-4f7d-8cb2-84fc45a90fde)

**Comment by gfourquet** [comment]
If you have a second shop (test, preprod)  with the same parameters, disable the module on this shops
That('s the recommandation from PS Support

**Comment by Temerario74** [comment]
No, no second shop.

**Comment by Guillaumelpg** [comment]
Hello,

For those of you who have had your store dissociated, we deployed a patch for the ps_accounts module yesterday in the afternoon.

We have ensured that stores that have been dissociated are automatically re-associated. If this is not the case, we invite you to re-associate your stores.

If you encounter an error again, please contact PrestaShop support via this contact form: https://help-center.prestashop.com/en/contact. 

Guillaume

**Comment by gfourquet** [comment]
Hello Guillaume,
Where can we find this patch ?

Guillaume

**Reply by Guillaumelpg** [reply]
Hello @gfourquet,

There was no correction in the code of the module because it was an infrastructure problem. So there is no need to update the module.

Guillaume
